docs/en/developer/create-web-component/VUE.md

```diff
@@ -1,137 +1,445 @@
+    3. [Maven](#maven)
+    4. [Gradle](#gradle)
+2. [Generate the Vue application](#2-generate-the-vue-application)
+3. [Edit the Vue application](#3-edit-the-vue-application)
+    1. [create gradle.properties](#create-gradleproperties)
+    2. [create build.gradle](#create-buildgradle)
+    3. [add Gradle wrapper (gradlew) to the project](#add-gradle-wrapper-to-project)
+    4. [rename HelloWorld.vue](#rename-helloworldvue)
+    5. [edit App.vue](#edit-appvue)
+    6. [edit package.json](#edit-packagejson)
+    7. [edit babel.config.js](#edit-babelconfigjs)
+4. [Assemble and deploy the Vue application](#4-assemble-and-deploy-the-vue-application)
+5. [Add the component into uPortal](#5-add-the-component-into-uportal)
+    1. [create a portlet-definition.xml](#create-a-portlet-definitionxml)
+    2. [edit portlet definition](#edit-portlet-definition)
+    3. [replace CDATA in portlet definition](#replace-cdata-in-portlet-definition)
+    4. [other options for portlet definition](#other-options-for-portlet-definition)
+       (don't forget to add permissions)
+    5. [add webjar to resource server](#add-webjar-to-resource-server)
+6. Select the component in uPortal (note that in the default uPortal-start
+it appears that you can only add to other tabs, not the default home tab,
+which may be locked.)
+
+## 1. Prerequisites
+#### Node.js
+If you don't have node installed there are several ways to do it. One
+way is to use Node Version Manager ([nvm](https://github.com/creationix/nvm)).
+
+``` bash
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
+```
 
-> The Progressive JavaScript Framework
+Install the latest Long Term Support (LTS) version of node (currently 10.15.1).
 
-<https://vuejs.org>
+``` bash
+nvm install node
+```
 
-## Advantages
+#### Vue CLI
+If you dont have Vue cli installed (Node must already be installed).
+``` bash
+npm install --global @vue/cli
+```
 
-- Vue has official support for web components
-- Vue components are compact, usually under 50kb apiece
+#### Maven
+Use the appropriate package manager for your OS. These instructions were
+tested with maven version 3.6.
 
-## Trade offs
+#### Gradle
+Use the appropriate package manager for your OS. These instructions were
+tested with gradle 5.2.
 
-- Vue does not embed the framework in the component
-  - this requires a secondary import
-  - can lead to version conflict
+## 2. Generate the Vue application
+Replace `{component-name}` with the desired name for the component.
 
-## Prerequisites
+``` bash
+vue create {component-name} --default
+```
 
-- [Git][]
-- [Node.js][] (using [NVM][] is recommended)
-- [Vue CLI][]
+Install dependencies for legacy browser support in the newly generated app.
+``` bash
+cd {component-name}
 
-## Step-by-Step Guide
+npm install --save-dev @babel/{cli,plugin-transform-runtime,preset-env}
+```
 
-Start by using the following commands to setup a new Web Component project in Vue.
+## 3. Edit the Vue application
```

Comment:
Note, that instead of adding the Gradle wrapper after these next few steps, you can run `gradle install` in this directory to get the wrapper and skeleton versions of gradle.properties and build.gradle.

Reply:
Also consider breaking this section into two, one being `Add Gradle support for Maven WebJars` or along those line, and the other `Edit Vue Application`

Reply:
👍 Agreed with with Benito to add this as a Note 📓 .

Reply:
@bjagg @cousquer I have moved the gradle setup to be under a different header. When I was running through the instructions our student tried doing this with 
$ gradle install
He ran into an issue with this because of the gradle version it installed. I would be happy to add the note, however I think it should come with a warning that you may run into an error due to versioning. 
Please let me know how you would like to proceed. 
Thanks

Reply:
> He ran into an issue with this because of the gradle version it installed.

Any specific on the issue, is there a bugfix that should be applied to the Gradle script(s)?

> I would be happy to add the note, however I think it should come with a warning that you may run into an error due to versioning.

Maybe we could/should add a minimum version plugin to warn when on older version of Gradle is used?
https://github.com/gradle/gradle/issues/2263